pyuvsim/simsetup.py

```diff
@@ -44,12 +45,84 @@ def _parse_layout_csv(layout_csv):
+        latitude_deg: Latitude of telescope in degrees. Used for declination coarse horizon cut.
+        horizon_buffer: Angle (float, in radians) of buffer for coarse horizon cut. Default is about 10 minutes of sky rotation.
+                        Sources whose calculated altitude is less than -horizon_buffer are excluded by the catalog read.
+                        Caution! The altitude calculation does not account for precession/nutation of the Earth. A decent buffer is needed to
+                        ensure that the horizon cut doesn't exclude sources near the horizon.
+        min_flux: Minimum flux to select [Jy]
+        max_flux: Maximum flux to select [Jy]
+    """
+
+    sourcelist = []
+    if len(catalog_table.shape) == 0:
+        catalog_table = catalog_table.reshape((1))
+    Nsrcs = catalog_table.shape[0]
+    coarse_kwds = [lst_array, latitude_deg]
+    coarse_horizon_cut = all([k is not None for k in coarse_kwds])
+    if (not coarse_horizon_cut) and any([k is not None for k in coarse_kwds]):
+        warnings.warn("It looks like you want to do a coarse horizon cut, but you're missing keywords")
+
+    if coarse_horizon_cut:
+        lat_rad = np.radians(latitude_deg)
+        buff = horizon_buffer
+    for (source_id, ra_j2000, dec_j2000, flux_I, freq) in catalog_table:
+        if min_flux:
+            if (flux_I < min_flux):
+                continue
+        if max_flux:
+            if (flux_I > max_flux):
+                continue
+        ra = Angle(ra_j2000, units.deg)
+        dec = Angle(dec_j2000, units.deg)
+        rise_lst = None
+        set_lst = None
+        if coarse_horizon_cut:
+            # Identify circumpolar sources and unrising sources.
+            tans = np.tan(lat_rad) * np.tan(dec.rad)
+            if tans < -1:
+                continue   # Source doesn't rise.
+            circumpolar = tans > 1
+
+            if not circumpolar:
+                rise_lst = ra.rad - np.arccos((-1) * tans)
+                set_lst = ra.rad + np.arccos((-1) * tans)
+                rise_lst -= buff
+                set_lst += buff
+
+                if rise_lst < 0:
+                    rise_lst += 2 * np.pi
+                if set_lst > 2 * np.pi:
+                    set_lst -= 2 * np.pi
+
+        source = Source(source_id, ra,
+                        dec,
+                        freq=freq * units.Hz,
+                        stokes=np.array([flux_I, 0., 0., 0.]),
+                        rise_lst=rise_lst,
+                        set_lst=set_lst)
+
+        sourcelist.append(source)
+
+    return np.array(sourcelist)
+
+
+def read_votable_catalog(gleam_votable, input_uv=None, source_select_kwds={}):
     """
     Creates a list of pyuvsim source objects from a votable catalog.
 
     Args:
         gleam_votable: Path to votable catalog file.
+        input_uv: The UVData object for the simulation (needed for horizon cuts)
+        source_select_kwds: Keywords for source selection (passed along)
```

Comment:
I'd prefer to define the relevant keywords on this function and then pass them through. Currently, they're only defined on a private function (starts with an underscore), which will not be documented on RTD (it skips private functions).

Reply:
I decided to make the 'array_to_sourcelist` function public. I can still define the options here if you think that'd be better.

Reply:
I think it is better. I know it involves copy/pasting but it's docs not code and I think people find it easier to read docs that are close to the code.